operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -229,8 +232,8 @@ type Rack struct {
 	// The rack name
 	// +kubebuilder:validation:MinLength=2
 	Name string `json:"name"`
-	// Zone name to pin the rack, using node affinity
-	Zone string `json:"zone,omitempty"`
```

Comment:
I would suggest naming the labels field `nodeAffinityLabels`, and we can't just delete the `zone` field here - because backwards compatibility

Reply:
@jimdickinson yeah I had it in mind. I can take Zone and add it in the nodeAffinityLabels when it's defined. I could probably also add a comment to mark **Zone** as deprecated